browser/ui/webui/brave_web_ui_controller_factory.cc

```diff
@@ -70,14 +71,15 @@ typedef WebUIController* (*WebUIFactoryFunction)(WebUI* web_ui,
 
 WebUIController* NewWebUI(WebUI* web_ui, const GURL& url) {
   auto host = url.host_piece();
+  Profile* profile = Profile::FromBrowserContext(
+      web_ui->GetWebContents()->GetBrowserContext());
   if (host == kAdblockHost) {
     return new BraveAdblockUI(web_ui, url.host());
   } else if (host == kWebcompatReporterHost) {
     return new WebcompatReporterUI(web_ui, url.host());
 #if BUILDFLAG(IPFS_ENABLED)
   } else if (host == kIPFSWebUIHost &&
-             ipfs::IsIpfsEnabled(
-                 web_ui->GetWebContents()->GetBrowserContext())) {
+             ipfs::IpfsServiceFactory::IsIpfsEnabled(profile)) {
```

Comment:
not sure this change is needed, it should work fine with the context as well?

Reply:
we split checks for browser and component level, and removed old ones, ipfs::IpfsServiceFactory::IsIpfsEnabled should be used on the browser side everywhere

Reply:
I mean there is no need to obtain a profile, BrowserContext should be enough.